crates/cac_client/src/lib.rs

```diff
@@ -188,14 +216,20 @@ impl Client {
         query_data: Map<String, Value>,
         merge_strategy: MergeStrategy,
     ) -> Result<Map<String, Value>, String> {
-        let cac = self.config.read().await;
-        eval::eval_cac(
-            cac.default_configs.to_owned(),
-            &cac.contexts,
-            &cac.overrides,
-            &query_data,
-            merge_strategy,
-        )
+        let context_key = json_to_sorted_string(&json!(query_data));
+        let value = self.config_cache.get(&context_key).unwrap_or({
+            let cac = self.config.read().await;
+            let evaled_val = eval::eval_cac(
+                cac.default_configs.to_owned(),
+                &cac.contexts,
+                &cac.overrides,
+                &query_data,
+                merge_strategy,
+            )?;
```

Comment:
`unwrap_or_else` would not work if you want to use early return from here
in that case you would have to convert this to a `match` block